config/initializers/ab_tests.rb

```diff
@@ -102,4 +102,14 @@ def self.all
         IdentityConfig.store.recommend_webauthn_platform_for_sms_ab_test_authentication_percent,
     },
   ).freeze
+
+  DESKTOP_FT_UNLOCK_SETUP = AbTest.new(
+    experiment_name: 'Desktop F/T unlock setup',
+    should_log: [
+      'WebAuthn Setup Visited',
+      'Multi-Factor Authentication Setup',
+    ].to_set,
```

Comment:
The validation metrics for this test require us to measure how many people successfully set up WebAuthn. I think we'll need to revise this list of events (e.g. "submitted" event rather than "visited").

It'd be a good idea to mock up the CloudWatch queries we expect to run to perform the metrics validation.

Reply:
The way we measure conversion rate is:

`# of successful submissions / # of users who ever visited the setup page`

Reply:
I think that's close, but we'll need to adjust to account for:

- Excluding non-English
- Deduplicate if someone visits 2FA setup twice in the same account creation flow for 2nd MFA upsell / forced 2nd MFA for Face/Touch
- There is no `success` property for `User Registration: 2FA Setup visited`

Reply:
I think the way you have `@success` considered in `fields` can be used for what you're trying to get at, but if we include it in `filter`, it's going to filter out "User Registration: 2FA Setup visited" events because those don't have a `success` property. Similarly, while `sum(@submitted and @success)` makes sense because submissions have `success`, `sum(@visited and @success)` should just be `sum(@visited)` because there is no `success` property.

Reply:
For deduplicating, we might be able to use CloudWatch's new support for multiple `stats` in the same query.

So something like... (untested)

```
filter name in ['Multi-Factor Authentication Setup', 'User Registration: 2FA Setup visited']
and properties.properties.browser_platform_name not in ['Android', 'iOS (iPhone)']
and properties.event_properties.in_account_creation_flow
and properties.ab_tests.desktop_ft_unlock_setup = 'desktop_ft_unlock_option_shown'
and properties.locale = 'en'
| fields (properties.event_properties.success) as @success,
(name = 'User Registration: 2FA Setup visited') as @visited,
(name = 'Multi-Factor Authentication Setup') as @submitted
| stats sum(@submitted and @success) > 0 as @successful_submit, sum(@visited) as @visited_once by properties.user_id
| stats sum(@successful_submit)/sum(@visited_once)*100 as `FT Unlock Conversion Rate` by bin(8h)
```

Reply:
I'm not sure we can use multiple `stats` on this query though because wouldn't `@visited_once` be determined by a timestamp and it looks like that field does not apply to anything related to time?

Reply:
Rather, the problem lies within the `bin(8h)`, but I am not sure why that is happening